Marlin/src/pins/ramps/pins_MKS_GEN_L.h

```diff
@@ -37,10 +39,30 @@
 // Heaters / Fans
 //
 
-#define MOSFET_A_PIN                          10  // HE0
-#define MOSFET_B_PIN                           7  // HE1 or FAN Hotend Cooling
-#define MOSFET_C_PIN                           8  // HBED
-#define FAN0_PIN                               9  // FAN Part Cooling
+#ifndef MOSFET_A_PIN
+  #define MOSFET_A_PIN                        10  // HE0
+#endif
+#ifndef MOSFET_B_PIN
+  #define MOSFET_B_PIN                         7  // HE1 or FAN Hotend Cooling
+#endif
+#ifndef MOSFET_C_PIN
+  #define MOSFET_C_PIN                         8  // H-BED
+#endif
+#ifndef MOSFET_D_PIN
+  #define MOSFET_D_PIN                         9  // FAN Part Cooling
+#endif
+
+// Always assign these to their labeled counterparts
+#define HEATER_0_PIN                MOSFET_A_PIN
+#if ANY(FET_ORDER_EEF, FET_ORDER_EEB)
+  #define HEATER_1_PIN              MOSFET_B_PIN
+#endif
+#define HEATER_BED_PIN              MOSFET_C_PIN
+
+#define FAN0_PIN                    MOSFET_D_PIN
+#if ENABLED(FET_ORDER_EFF)
+  #define FAN1_PIN                  MOSFET_B_PIN
+#endif
```

Comment:
This should do better, and is more complete. Boards like this one that include `pins_RAMPS.h` and have a preferred plug ordering should all follow this sort of approach and override the standard RAMPS assignments.

While many boards do not implement assignment of the PWM pins based on the `FET_ORDER_*` flags, all pins files are free to do so to handle standard rearrangements and fit them to their preferred usage.